### PR TITLE
Don't revert Jira status if LP hasn't actually reverted

### PR DESCRIPTION
**Pull request by schopin-pro**
There are some cases where the actual work will be done but the LP
status of the bug will not move, as the LP status has specific semantics
tied to a published package in the archive, and in some cases the work
will only be done in a Git repo but will wait for a later bulk upload to
be reflected in any published package.

For example, this frequently happens for systemd and glibc.

In addition, when working on an SRU, the uploaded package can be stuck
for a while in the SRU queue before publication, and can also sometimes
take a while before released into the release pocket.

Allowing a Jira issue to move to Done and stay there while the LP task
is still in one of those states makes working on those tasks much
easier, without having to remember to remove the relevant tag from the
issue.

### CODE REVIEW

**Review by mclemenceau — CHANGES_REQUESTED** [review]
Thanks for the contribution unfortunately, it didn't work with my local testing.
I will need to do better comments in the future to prevent this as some data structure may be somewhat obscure.
Also thanks for the typing changes, I'm too familiar with this so I will look into that further.
Bottom line, this logic could work but we may have to make a call about which task to prioritize among the possible series affected. Ideally we can identify the devel serie and ignore the other status.
It appears we have a function lp_status that isn't being used that could probably be changed to do what we actually want here.

File: lp_to_jira_sync/lp_to_jira_sync.py [hunk 1]
```diff
@@ -367,8 +370,28 @@ def log(msg):
     if jira_comment:
         config.jira.add_comment(issue, jira_comment)
 
+def revert_jira_status(config: SyncConfig, jira_issue: Issue, task: Any):
+    if task.status in (
```
[the remark this answers is not in the document]
Comment (schopin-pro) [reply lp_to_jira_sync/lp_to_jira_sync.py]: I pushed a fix that should work, under the assumptions that all tasks in a given `all_task` item are about the relevant package.

File: lp_to_jira_sync/lp_to_jira_sync.py [hunk 2]
```diff
@@ -367,8 +370,28 @@ def log(msg):
...
+        'anymore, please untag the bug in LP.') % (config.tag)
+    if not config.dry_run:
+        config.jira.transition_issue(
+            jira_issue,
+            transition='Triaged'
+        )
+        config.jira.add_comment(jira_issue, comment)
+
+
+def process_issues(all_tasks: dict[Bugset, Any], all_issues: dict[Bugset, Issue], config):
```
[the remark this answers is not in the document]
Comment (schopin-pro) [reply lp_to_jira_sync/lp_to_jira_sync.py]: Fixed, thanks!

File: tests/test_lp_to_jira_sync.py [hunk 3]
```diff
@@ -22,3 +22,30 @@ def test_get_bug_pkg():
...
+    config = MagicMock(tag="bogus-tag", dry_run=False)
+    tasks = [
+            MagicMock(status="Fix Released"), # already fixed on devel series
+            MagicMock(status="In Progress"), # SRU in queue for last stable
+            MagicMock(status="Won't Fix"), # Not planning on fixing almost EOL interim release
+            MagicMock(status="In Progress"), # SRU in queue for last LTS
+            ]
+    issue = MagicMock(id="FR-1234")
+
+    breakpoint()
```
Comment (mclemenceau) [inline tests/test_lp_to_jira_sync.py:36]: Can you remove that breakpoint ?

**Review by mclemenceau — CHANGES_REQUESTED** [review]
Working as expected, lightly tested with actual Jira project. Note that this would require all the subtasks to be in Progress states or it would be moved back to Triaged
Other than that breakpoint() to be removed, I'm happy with it

File: tests/test_lp_to_jira_sync.py (hunk 3 above)
Comment (schopin-pro) [reply tests/test_lp_to_jira_sync.py]: Silly me!

**Pull request merged**